src/Compiler/TypedTree/TcGlobals.fsi

```diff
@@ -0,0 +1,1377 @@
+    member memoize_file: x: int -> FSharp.Compiler.AbstractIL.IL.ILSourceDocument
+
+    member mkDebuggableAttributeV2:
+        jitTracking: bool * jitOptimizerDisabled: bool -> FSharp.Compiler.AbstractIL.IL.ILAttribute
+
+    member mkDebuggerDisplayAttribute: s: string -> FSharp.Compiler.AbstractIL.IL.ILAttribute
+
+    member mk_ArrayCollector_ty: seqElemTy: FSharp.Compiler.TypedTree.TType -> FSharp.Compiler.TypedTree.TType
+
+    member mk_GeneratedSequenceBase_ty: seqElemTy: FSharp.Compiler.TypedTree.TType -> FSharp.Compiler.TypedTree.TType
+
+    member mk_IResumableStateMachine_ty: dataTy: FSharp.Compiler.TypedTree.TType -> FSharp.Compiler.TypedTree.TType
+
+    member mk_ListCollector_ty: seqElemTy: FSharp.Compiler.TypedTree.TType -> FSharp.Compiler.TypedTree.TType
+
+    member mk_ResumableStateMachine_ty: dataTy: FSharp.Compiler.TypedTree.TType -> FSharp.Compiler.TypedTree.TType
+
+    member tryRemoveEmbeddedILTypeDefs: unit -> FSharp.Compiler.AbstractIL.IL.ILTypeDef list
+
+    member unionCaseRefEq:
+        x: FSharp.Compiler.TypedTree.UnionCaseRef -> y: FSharp.Compiler.TypedTree.UnionCaseRef -> bool
+
+    member valRefEq: x: FSharp.Compiler.TypedTree.ValRef -> y: FSharp.Compiler.TypedTree.ValRef -> bool
+
+    member CompilerGeneratedAttribute: FSharp.Compiler.AbstractIL.IL.ILAttribute
+
+    member CompilerGlobalState: FSharp.Compiler.CompilerGlobalState.CompilerGlobalState option
+
+    member DebuggerBrowsableNeverAttribute: FSharp.Compiler.AbstractIL.IL.ILAttribute
+
+    member DebuggerNonUserCodeAttribute: FSharp.Compiler.AbstractIL.IL.ILAttribute
+
+    member IComparer_ty: FSharp.Compiler.TypedTree.TType
+
+    member IEqualityComparer_ty: FSharp.Compiler.TypedTree.TType
+
+    member ListCollector_tcr: FSharp.Compiler.TypedTree.EntityRef
+
+    member MatchFailureException_tcr: FSharp.Compiler.TypedTree.EntityRef
+
+    member ResumableCode_tcr: FSharp.Compiler.TypedTree.EntityRef
+
+    member System_Runtime_CompilerServices_RuntimeFeature_ty: FSharp.Compiler.TypedTree.TType option
+
+    member addrof2_vref: FSharp.Compiler.TypedTree.ValRef
+
+    member addrof_vref: FSharp.Compiler.TypedTree.ValRef
+
+    member and2_vref: FSharp.Compiler.TypedTree.ValRef
+
+    member and_vref: FSharp.Compiler.TypedTree.ValRef
+
+    member array2D_get_info: IntrinsicValRef
+
+    member array2D_get_vref: FSharp.Compiler.TypedTree.ValRef
+
+    member array2D_set_info: IntrinsicValRef
+
+    member array3D_get_info: IntrinsicValRef
+
+    member array3D_get_vref: FSharp.Compiler.TypedTree.ValRef
+
+    member array3D_set_info: IntrinsicValRef
+
+    member array4D_get_info: IntrinsicValRef
+
+    member array4D_get_vref: FSharp.Compiler.TypedTree.ValRef
+
+    member array4D_set_info: IntrinsicValRef
+
+    member array_get_info: IntrinsicValRef
+
+    member array_get_vref: FSharp.Compiler.TypedTree.ValRef
+
+    member array_length_info: IntrinsicValRef
+
+    member array_map_info: IntrinsicValRef
```

Comment:
I am removing `array_map*` and `list_map*` in #17421 (which is ready to review, incidentally).